target/scripts/helpers/postfix.sh

```diff
@@ -92,39 +92,42 @@ function _vhost_ldap_support() {
 # /etc/aliases is handled by `alias.sh` and uses `postalias` to update the Postfix alias database. No need for `postmap`.
 # http://www.postfix.org/postalias.1.html
 
-# Add an key with an value to Postfix's main configuration file
+# Add a key with a value to Postfix's main configuration file
 # or update an existing key. An already existing key can be updated
 # by either appending to the existing value (default) or by prepending.
 #
 # @param ${1} = key name in Postfix's main configuration file
 # @param ${2} = new value (appended or prepended)
-# @param ${3} = "append" (default) or "prepend" [OPTIONAL]
+# @param ${3} = action "append" (default) or "prepend" [OPTIONAL]
 function _add_to_or_update_postfix_main() {
   local KEY=${1:?Key name is required}
   local NEW_VALUE=${2:?New value is required}
   local ACTION=${3:-append}
+  local CURRENT_VALUE
 
-  # If entry does not exist, add it - otherwise update with ACTION:
-  if ! grep -q -E "^${KEY}" /etc/postfix/main.cf; then
+  # Get current value from /etc/postfix/main.cf
+  CURRENT_VALUE=$(postconf -h "${KEY}" 2>/dev/null)
+
+  # If key does not exist or value is empty, add it - otherwise update with ACTION:
+  if [[ -z ${CURRENT_VALUE} ]]; then
     postconf "${KEY} = ${NEW_VALUE}"
   else
-    KEY=$(_escape_for_sed "${KEY}")
-    NEW_VALUE=$(_escape_for_sed "${NEW_VALUE}")
-    local SED_STRING
+    # If $NEW_VALUE is already present --> nothing to do, skip.
+    if grep -qF " ${NEW_VALUE} " <<<" ${CURRENT_VALUE} "; then
```

Comment:
Pure BASH should also be fine?
```suggestion
    if [[ " ${CURRENT_VALUE} " == *" ${NEW_VALUE} "* ]]; then
```

Reply:
I am not a bash wizard like you two 😅 

Despite the glory of DMS bash scripts, we've not attracted anymore wizards as maintainers, so I'd probably go with the more familiar `grep` pattern from a maintenance perspective?

Reply:
I'm not sure how reliable this condition logic is though in practice.

- ~~I know you're using white-space to avoid accidents where value could belong to a prefix/suffix portion of something else.~~

...Nevermind! Being the smart wizard that you are, you're not using `grep` this way on `main.cf` contents which would have been riddled with a few gotchas 😅 (_I should probably do reviews once I've had a morning coffee_ 😝 )

Reply:
I also prefer `[[...` over `grep ...`.

Reply:
Switched to pure bash. Using build-ins should also a bit faster 😆 than calling external programs like `grep`.